src/core/Akka.Remote.Tests/ActorsLeakSpec.cs

```diff
@@ -0,0 +1,220 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Akka.Actor;
+using Akka.Actor.Internal;
+using Akka.Configuration;
+using Akka.Remote.Transport;
+using Akka.TestKit;
+using Akka.TestKit.TestActors;
+using Akka.Util.Internal;
+using Xunit;
+
+namespace Akka.Remote.Tests
+{
+    public class ActorsLeakSpec : AkkaSpec
```

Comment:
This spec is pretty genius - it maps the system remoting actor heirarchies and verifies that after all of the scenarios below that we haven't accidentally "leaked" any remoting actors; that they're always cleaned up properly after disassociations under all of these scenarios.

This spec revealed two bugs with the `ReliableDeliverySupervisor` and the `EndpointManager`